packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -485,6 +485,55 @@ export class Key extends KeyBase {
     return new Import(keyResourceName);
   }
 
+  /**
+   * Create a mutable {@link IKey} based on a low-level {@link CfnKey}.
+   * This is most useful when combined with the cloudformation-include module.
+   * This method is different than {@link fromKeyArn()} because the {@link IKey}
+   * returned from this method is mutable;
+   * meaning, calling any mutating methods on it,
+   * like {@link IKey.addToResourcePolicy()},
+   * will actually be reflected in the resulting template,
+   * as opposed to the object returned from {@link fromKeyArn()},
+   * on which calling those methods would have no effect.
+   */
+  public static fromCfnKey(cfnKey: CfnKey): IKey {
+    // use a "weird" id that has a higher chance of being unique
+    const id = '@FromCfnKey';
+
+    // if fromCfnKey() was already called on this cfnKey,
+    // return the same L2
+    // (as different L2s would conflict, because of the mutation of the keyPolicy property of the L1 below)
+    const existing = cfnKey.node.tryFindChild(id);
+    if (existing) {
```

Comment:
I think if a user created a different resource with the weird ID, we're screwed anyway, so I'd rather not add this check 😜 (especially since there can be implementations of `IKey` that do not extend `KeyBase`, and they should be able to interoperate with these classes freely).

Reply:
Replace KeyBase with IKey and I think there is still a problem. The user would think that their promotion worked but if they tried to use it in, for example, an S3 bucket, then we would silently produce an incorrect template (missing kmsMasterKeyId since there would be no property `keyArn` on the object).

Reply:
While what you're saying is  technically true, I think this falls in "pragmatically okay to do make some assumptions" territory.

It's unlikely that a user is *accidentally* going to create a construct named `@FromCfnKey`, and if they're doing it on purpose then they might have good reason to do it.